pmd-core/src/main/java/net/sourceforge/pmd/renderers/internal/sarif/SarifLog.java

```diff
@@ -588,15 +595,25 @@ public int hashCode() {
                 return this;
             }
 
+            /**
+             * A set of distinct strings that provide additional information. This is SARIF 2.1.0 Schema.
+             * @return {@code this}.
+             */
+            @java.lang.SuppressWarnings("all")
+            public SarifLog.PropertyBag.PropertyBagBuilder tags(final String[] tags) {
+                this.tags = tags;
+                return this;
+            }
+
             @java.lang.SuppressWarnings("all")
             public SarifLog.PropertyBag build() {
-                return new SarifLog.PropertyBag(this.ruleset, this.priority);
+                return new SarifLog.PropertyBag(this.ruleset, this.priority, this.tags);
             }
 
             @java.lang.Override
             @java.lang.SuppressWarnings("all")
             public java.lang.String toString() {
-                return "SarifLog.PropertyBag.PropertyBagBuilder(ruleset=" + this.ruleset + ", priority=" + this.priority + ")";
+                return "SarifLog.PropertyBag.PropertyBagBuilder(ruleset=" + this.ruleset + ", priority=" + this.priority + ", tags=" + this.tags + ")";
```

Comment:
this is for instance one of the unpleasant things with arrays, the toString will print `[Ljava.lang.String;@1234567`